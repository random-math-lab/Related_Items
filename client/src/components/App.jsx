import React from 'react';
import axios from 'axios';
import RelatedListings from './relatedListings.jsx'

class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            currentListing: [],
        }
    }

    componentDidMount() {
        this.getData();
    }

    getData() {
        axios('/api/relatedPlaces/1')
        .then( (res) => res.data )
        .then( (places) => this.setState({currentListing : places}) )
        // fetch('/api/relatedPlaces/1')    
        // .then( (data) =>  (data.json()) )
        // .then( (message) => JSON.stringify(message) )
        // .then( (string) => JSON.parse(string) )
        // .then( (output) => this.setState( {currentListing: output} ) )
    }
    

    render() {
        if(this.state.currentListing.length) {
            return (
                <RelatedListings listings={this.state.currentListing}/>
            )
        }
        return (

            <div>
            </div>
        )
    }

}

export default App;