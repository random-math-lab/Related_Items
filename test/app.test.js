// import seeder from '.../database/seeder.js'

import React from 'react';
import { shallow } from 'enzyme';
import App from '../client/src/components/App.jsx';


describe('testing app', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App />);
  
    expect(component).toMatchSnapshot();
  });
});