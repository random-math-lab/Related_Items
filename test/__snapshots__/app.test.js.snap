// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing app should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
